packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -111,21 +144,53 @@ class FlutterPlugin implements Plugin<Project> {
         return result
     }
 
+    /**
+     * Returns the platform that is used to extract the `libflutter.so` and the .class files.
```

Comment:
It extracts `libflutter.so` as well, then it adds the `libflutter.so` for the rest of the ABIs.
Alternatively, we could extract the `.class` only before the `afterEvaluate` block (otherwise Gradle complains that we are referencing classes that aren't defined).

I filed an issue to track this in https://github.com/flutter/flutter/issues/33059 and https://github.com/flutter/flutter/issues/33064

The current engine artifacts for Android are:
```
android-arm/
android-arm-profile/
android-arm-release/
android-arm64/
android-arm64-profile/
android-arm64-release/
android-x64/
android-x86/
```

`x64` and `x86` are for debug only, and their corresponding `libflutter.so` is always added even for `arm` and `arm64`.